libraries/Pliant/Grammars/Frame.cs

```diff
@@ -1,11 +1,26 @@
 ﻿using Pliant.Tokens;
 using Pliant.Utilities;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace Pliant.Grammars
 {
     internal class Frame
     {
+        private PreComputedState[] _cachedData;
```

Comment:
Same deal here as with the PreComputedState cached frames. We could reuse the ArrayWrapper&lt;T&gt;